### PR TITLE
Feature/03 homepage

### PR DESCRIPTION
**Pull request by Siiddy**

### CODE REVIEW

File: eslint.config.mjs
Comment (lchaufournais) [inline eslint.config.mjs:1]: Need to delete this file.

File: jsconfig.json
Comment (lchaufournais) [inline jsconfig.json:1]: Why it's empty ?

File: public/images/Instamint.png
Comment (lchaufournais) [inline public/images/Instamint.png:1]: Use bucket

File: public/images/mint.png
Comment (lchaufournais) [inline public/images/mint.png:1]: Use bucket

File: public/images/mintSolid.png
Comment (lchaufournais) [inline public/images/mintSolid.png:1]: Use bucket

File: src/api-post/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import prepareRoutes from "./prepareRoutes.js"
+
+const db = knex(config.db)
+BaseModel.knex(db)
+
+const app = new Hono()
+app.use(logger())
+app.use(
+  cors({
+    origin: "*",
```
Comment (lchaufournais) [inline src/api-post/index.js]: Last correction from Avetis said that we can't allow all.

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { Hono } from "hono"
+import { z } from "zod"
+import { zValidator } from "@hono/zod-validator"
+import { idValidator } from "../validators.js"
+
+const prepareRoutesLike = ({ app, db }) => {
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Why use db ?

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    postId: idValidator,
+    userId: idValidator.optional()
+  })
+
+  likeData.get("/post/liked/:postId/:userId", zValidator("param", likeSchema), async c => {
+    const postId = c.req.valid("param").postId
+    const userId = c.req.valid("param").userId
+
+    try {
+      const existingLike = await db.table("likes").where({ postId, userId }).first()
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Use your model

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+      return c.json({ success: false, message: `Error checking like status for post ${postId}: ${error.message}` })
+    }
+  })
+
+  likeData.get("/post/likes/:postId/:userId", zValidator("param", likeSchema), async c => {
+    const postId = c.req.valid("param").postId
+
+    try {
+      const likeCount = await db.table("likes").where({ postId }).count("* as likeCount")
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Use your model

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+      return c.json({ success: false, message: `Error retrieving like count for post ${postId}: ${error.message}` })
+    }
+  })
+
+  likeData.post("/post/like/:postId/:userId", zValidator("param", likeSchema), async c => {
+    const postId = c.req.valid("param").postId
+    const userId = c.req.valid("param").userId
+
+    const existingLike = await db.table("likes").where({ postId, userId }).first()
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Use your model

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    await db.table("likes").insert({ postId, userId })
+
+    return c.json({ success: true, message: "Like added to post" })
+  })
+
+  likeData.delete("/post/like/:postId/:userId", zValidator("param", likeSchema), async c => {
+    const postId = c.req.valid("param").postId
+    const userId = c.req.valid("param").userId
+
+    const existingLike = await db.table("likes").where({ postId, userId }).first()
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Use Your model

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+    if (!existingLike) {
+      c.status(404)
+
+      return c.json({ success: false, message: "You have not liked this post" })
+    }
+
+    await db.table("likes").where({ postId, userId }).delete()
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Use your model

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  likeData.get("/post/liked/:postId/:userId", zValidator("param", likeSchema), async c => {
+    const postId = c.req.valid("param").postId
+    const userId = c.req.valid("param").userId
+
+    try {
+      const existingLike = await db.table("likes").where({ postId, userId }).first()
+
+      return c.json({ success: true, isLiked: Boolean(existingLike) })
+    } catch (error) {
+      c.status(500)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+  likeData.get("/post/likes/:postId/:userId", zValidator("param", likeSchema), async c => {
+    const postId = c.req.valid("param").postId
+
+    try {
+      const likeCount = await db.table("likes").where({ postId }).count("* as likeCount")
+
+      return c.json({ success: true, postId, likeCount: likeCount[0].likeCount })
+    } catch (error) {
+      c.status(500)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/api-post/src/routes/posts/LikeRoute.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  })
+
+  likeData.delete("/post/like/:postId/:userId", zValidator("param", likeSchema), async c => {
+    const postId = c.req.valid("param").postId
+    const userId = c.req.valid("param").userId
+
+    const existingLike = await db.table("likes").where({ postId, userId }).first()
+
+    if (!existingLike) {
+      c.status(404)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/LikeRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/api-post/src/routes/posts/SubscribedRoute.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    "/post/subscribed/:id",
+    zValidator("param", idSubscribedSchema),
+    zValidator("query", pageSubscribedSchema),
+    async c => {
+      try {
+        const userId = c.req.valid("param").id
+        const page = parseInt(c.req.valid("query").page) || 0
+
+        if (page < 0) {
+          c.status(400)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/SubscribedRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/api-post/src/routes/posts/SubscribedRoute.js [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+          return c.json({
+            success: false,
+            message: "Invalid page number"
+          })
+        }
+
+        const followedUsers = await FollowModel.query().where("followerId", userId)
+
+        if (followedUsers.length === 0) {
+          c.status(404)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/SubscribedRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/api-post/src/routes/posts/SubscribedRoute.js [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+          return c.json({
+            success: false,
+            message: "No followed users"
+          })
+        }
+
+        const followedIds = followedUsers.map(user => user.followedId)
+
+        if (followedIds.length === 0) {
+          c.status(404)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/SubscribedRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/api-post/src/routes/posts/SubscribedRoute.js [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+            "posts.createdAt",
+            "posts.description",
+            "posts.imageUrl",
+            "users.username",
+            "users.profileImage"
+          )
+          .orderBy("posts.createdAt", "desc")
+
+        if (followedPosts.length === 0) {
+          c.status(404)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/SubscribedRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/api-post/src/routes/posts/SubscribedRoute.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+
+          return c.json({
+            success: false,
+            message: "No followed posts"
+          })
+        }
+
+        c.status(200)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/SubscribedRoute.js]: Last correction said that we can't use magic number to declare our http code.

File: src/api-post/src/routes/posts/SubscribedRoute.js [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+
+        return c.json({
+          success: true,
+          result: followedPosts,
+          hasMore: followedPosts.length > 10
+        })
+      } catch (error) {
+        c.status(500)
```
Comment (lchaufournais) [inline src/api-post/src/routes/posts/SubscribedRoute.js]: Last correction said that we can't use magic number to declare our error code.

File: src/app/components/business/FeedPost.jsx [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+
+  useEffect(() => {
+    const fetchPosts = async () => {
+      try {
+        let apiError, apiData
+
+        if (type === "forYou") {
+          ;[apiError, apiData] = await getForyouPost({ userId: session.id, page })
+        } else if (type === "subscribed") {
+          ;[apiError, apiData] = await getSubscribedPost({ userId: session.id, page })
```
Comment (lchaufournais) [inline src/app/components/business/FeedPost.jsx]: Syntax error

File: src/app/components/business/FeedPost.jsx [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+  const [page, setPage] = useState(0)
+  const [error, setError] = useState(null)
+
+  useEffect(() => {
+    const fetchPosts = async () => {
+      try {
+        let apiError, apiData
+
+        if (type === "forYou") {
+          ;[apiError, apiData] = await getForyouPost({ userId: session.id, page })
```
Comment (lchaufournais) [inline src/app/components/business/FeedPost.jsx]: Syntax error

File: src/app/hooks/usePostInteractions.jsx
Comment (lchaufournais) [inline src/app/hooks/usePostInteractions.jsx:1]: I don't really get why you do this but why not

**Review by lchaufournais — CHANGES_REQUESTED** [review]

File: src/pages/home/index.jsx [hunk 19]
```diff
@@ -1,13 +1,36 @@
...
+import Footer from "../../app/components/common/Footer.jsx"
+import Header from "../../app/components/common/Header.jsx"
+import { PaperAirplaneIcon } from "@heroicons/react/24/outline"
+// import useAppContext from "@/app/hooks/useContext.jsx"
 
 const Home = () => {
+  // This is not dead code, we only comment it out to avoid making too many requests to the bucket because we are limited to 20,000 requests in the free version
+  // const {
+  //   action: { getImagesBucket }
+  // } = useAppContext()
```
Comment (avetisk) [inline src/pages/home/index.jsx:13]: Dead code!

Remove it.

File: src/app/styles/Home.module.css [hunk 20]
```diff
@@ -32,70 +32,102 @@
...
-	width: 60px;
-	height: 60px;
-	position: relative;
-	transform-origin: center;
-	animation: spin 3s infinite linear;
+  width: 60px;
+  height: 60px;
+  position: relative;
+  transform-origin: center;
+  animation: spin 3s infinite linear;
```
Comment (avetisk) [inline src/app/styles/Home.module.css:39]: Use tailwind, not plain CSS.

File: src/app/services/posts/likesPostGet.mjs [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+      throw new Error("apiClients is not defined yet")
+    }
+
+    const { apiPost } = apiClients
+
+    if (!apiPost || typeof apiPost.get !== "function") {
+      throw new Error("apiPost is not defined or not a function")
+    }
+
+    postId = parseInt(postId)
```
Comment (avetisk) [inline src/app/services/posts/likesPostGet.mjs:16]: You shouldn't need to do it here.

File: src/app/hooks/usePostInteractions.jsx [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        setIsLiked(true)
+        setLikeCount(prev => (prev != null ? parseInt(prev) + 1 : 1))
+      }
+    } catch (err) {
+      setError("Failed to update like status")
+      showError("Failed to update like status")
+    }
+  }, [postId, isLiked, deleteLikePost, postLikePost, session.id])
```
Comment (avetisk) [inline src/app/hooks/usePostInteractions.jsx:74]: Early returns are missing~!

**Pull request merged**